tools/src/dbgen/Lib.hs

```diff
@@ -172,21 +202,85 @@ generateWalletDB CLI{..} spec@GenSpec{..} = do
 
     -- TODO(ks): Simplify this?
     let fakeUtxoSpec = fakeUtxoCoinDistr walletSpec
+    let fakeTxs      = fakeTxsHistory walletSpec
+
     case addTo of
-        Just accId -> case fakeUtxoSpec of
-            NoDistribution -> addAddressesTo accId spec
-            _              -> generateFakeUtxo fakeUtxoSpec accId
+        Just accId ->
+            if (checkIfAddTo fakeUtxoSpec fakeTxs) then
+                addAddressesTo spec accId
+            else do
+                void $ generateFakeUtxo fakeUtxoSpec accId
```

Comment:
Why `void`?

Reply:
Ugh. Because I'm not looking at the function types? Will fix, good catch.

Reply:
Ugh. Because I'm not looking at function types? Will fix, nice catch.